src/calendar/tests/CalendarDate.spec.js

```diff
@@ -0,0 +1,463 @@
+        });
+      });
+
+      it('when the isHighlightedDate prop is passed in', function () {
+        this.useShallowRenderer({
+          isHighlightedDate: true,
+        });
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: jasmine.any(String),
+          modifiers: jasmine.any(Object),
+          states: {disabled: jasmine.any(Boolean), highlighted: true, selected: jasmine.any(Boolean)},
+        });
+      });
+
+      it('when the isHighlightedDate prop is not passed in', function () {
+        this.useShallowRenderer({
+          isHighlightedDate: false,
+        });
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: jasmine.any(String),
+          modifiers: jasmine.any(Object),
+          states: {disabled: jasmine.any(Boolean), highlighted: false, selected: jasmine.any(Boolean)},
+        });
+      });
+
+      it('when the isSelectedDate prop is passed in', function () {
+        this.useShallowRenderer({
+          isSelectedDate: true,
+        });
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: jasmine.any(String),
+          modifiers: jasmine.any(Object),
+          states: {disabled: jasmine.any(Boolean), highlighted: jasmine.any(Boolean), selected: true},
+        });
+      });
+
+      it('when the isInSelectedRange prop is passed in', function () {
+        this.useShallowRenderer({
+          isInSelectedRange: true,
+        });
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: jasmine.any(String),
+          modifiers: jasmine.any(Object),
+          states: {disabled: jasmine.any(Boolean), highlighted: jasmine.any(Boolean), selected: true},
+        });
+      });
+
+      it('when the isInHighlightedRange prop is passed in', function () {
+        this.useShallowRenderer({
+          isInHighlightedRange: true,
+        });
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: jasmine.any(String),
+          modifiers: jasmine.any(Object),
+          states: {disabled: jasmine.any(Boolean), highlighted: jasmine.any(Boolean), selected: true},
+        });
+      });
+
+      it('when the isSelectedDate, isInSelectedRange, isInHighlightedRange props are not passed in', function () {
+        this.useShallowRenderer({
+          isSelectedDate: false,
+          isInSelectedRange: false,
+          isInHighlightedRange: false,
+        });
+        expect(this.spyCx).toHaveBeenCalledWith({
+          element: jasmine.any(String),
+          modifiers: jasmine.any(Object),
+          states: {disabled: jasmine.any(Boolean), highlighted: jasmine.any(Boolean), selected: false},
+        });
+      });
+
+    });
+
+  });
+
+  describe('creates the right style', function () {
+
+    it('when numStyles is 1', function () {
+      this.useShallowRenderer();
+      expect(this.renderedComponent.props.style.borderLeftColor).toEqual('#29');
```

Comment:
I'm a fan of using one `expect` per `it`, with any set up being done in `beforeEach` blocks. I'm not sure what house style is though.

Do you share any thoughts / preferences @jkimbo ?